File: test/_harnesses/rust/src/broken.rs
// message files will include code that's not necessarily called in each test
#![allow(dead_code)]

use std::fs;

// in a real program you would want this file to have an appropriate name
//   but for testing I'd rather not deal with it having a different name
//   than all the other files.
#[allow(non_snake_case)]
mod BrokenMessages;
use BrokenMessages::*;

mod util;

fn main() {
    let mut checker = util::Checker { ok: true };
    let args = util::arg_parse();

    let broken = TruncatedMessage { x: 1.0, y: 2.0 };

    if args.contains_key("generate") {
        let mut writer: Vec<u8> = Vec::new();
        broken.write_bytes(&mut writer, false);

        let filename = args.get("generate").unwrap();
        checker.soft_assert(writer.len() == 8, "size calculation check");
        fs::write(filename, writer).unwrap();
    }
    else if args.contains_key("read") {
        let filename = args.get("read").unwrap();
        let buffer = fs::read(&filename).unwrap();
        let mut reader = BufferReader::new(buffer);
        let read_res = FullMessage::from_bytes(&mut reader);
        match read_res {
            Ok(_) => checker.soft_assert(false, "reading broken message"),
            Err(_) => checker.soft_assert(true, "reading broken message")
        }
    }

    checker.check();
}

File: test/_harnesses/assemblyscript/assembly/broken.ts
import * as harness from "./_harness";
export * from "./_harness";

import * as BrokenMessages from '../../../../out/generated/assemblyscript/BrokenMessages';

const broken = new BrokenMessages.TruncatedMessage();
broken.x = 1.0;
broken.y = 2.0;

export function generate(): usize {
    const byteLen = broken.getSizeInBytes();
    harness.allocate(byteLen);
    const dv = harness.getDataView();
    const success = broken.writeBytes(dv, false);
    harness.softAssert(success == true, "writing broken message");

    return harness.getDataPtr();
}

export function read(): void {
    const dv = harness.getDataView();
    const input = BrokenMessages.FullMessage.fromBytes(dv);

    harness.softAssert(input == null, "reading broken message");
}
